### ISSUE TITLE
Path parameters with explicit types are not correct at runtime

### ISSUE THREAD

**Issue by epotter2297** [ecyrbe/zodios#182]
If I have the following API and endpoint:
```ts
const userApi = makeApi([
  {
    method: 'get',
    path: '/users/:id',
    alias: 'getUser',
    description: 'Get a user',
    parameters: [
      {
        name: 'id',
        type: 'Path',
        schema: z.number(),
      },
    ],
    response: z.object({
      id: z.number(),
      name: z.string().nullable(),
    }),
    errors,
  },
]);

const userRouter = ctx.router(userApi);

userRouter.get('/users/:id', async (req, res) => {
  console.log(typeof req.params.id);

  return res.json({
    id: req.params.id,
    name: "example",
  });
});
```
The output of the `console.log()` is `string`, even though the type is inferred as `number`.

**Comment by ecyrbe** [comment]
@epotter2297 this is now fixed on `@zodios/express` v10.0.2

**Comment by ecyrbe** [comment]
@epotter2297 one thing was missing (boolean coercion), fixed on `@zodios/express` v10.0.3
